relay-server/src/services/processor.rs

```diff
@@ -499,6 +499,30 @@ impl<'a> ProcessEnvelopeState<'a> {
         self.remove_event();
         self.managed_envelope.reject_event(outcome);
     }
+
+    fn forward_unsampled_profiles(&self, global_config: &GlobalConfig) -> bool {
+        let Some(global_options) = global_config.options else {
+            return false;
+        };
+
+        if !global_options.unsampled_profiles_enabled {
+            return false;
+        }
+
+        let event_platform = self
+            .event
+            .value()
+            .and_then(|e| e.platform.as_str())
+            .unwrap_or("");
+
+        self.project_state
+            .has_feature(Feature::IngestUnsampledProfiles)
+            && global_options
+                .profile_metrics_allowed_platforms
+                .iter()
+                .any(|s| s == event_platform)
+            && rand::random::<f32>() < global_options.profile_metrics_sample_rate
```

Comment:
@viglia I did not know how strict this check should be, so I defaulted to strictest possible (AND-condition). It's up to you to decide how you want to handle this (e.g. should a project with the feature flag enabled respect the sample rate? Or is the sample rate just for projects that do not have the feature flag enabled? Etc.)

Reply:
@jjbayer, that's correct. It should be the strictest possible.

So we want: 
1. global_options.unsampled_profiles_enabled == True
2. feature flag enabled
3. platform being allowed
4. sampling